elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSinkStage.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.elasticsearch
+
+import java.util
+
+import akka.Done
+import akka.stream._
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler}
+import org.apache.http.entity.StringEntity
+import org.elasticsearch.client.RestClient
+import spray.json._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Future, Promise}
+
+final case class ElasticsearchSinkSettings(bufferSize: Int = 10)
+
+final case class IncomingMessage[T](id: Option[String], source: T)
```

Comment:
Hmm... In my opinion, holding id and object (which is converted to document json) separately is natural when using Elasticsearch. It's possible to replace with tuple like `(Option[String], T)`.